purchases/src/androidTestIntegrationTest/kotlin/com/revenuecat/purchases/PurchasesIntegrationTest.kt

```diff
@@ -164,18 +166,23 @@ class PurchasesIntegrationTest {
     fun canPurchaseSubsProduct() {
         val lock = CountDownLatch(1)
 
-        val storeProduct = StoreProductFactory.createStoreProduct()
+        val storeProduct = StoreProductFactory.createGoogleStoreProduct()
         val storeTransaction = StoreTransactionFactory.createStoreTransaction()
-        mockBillingAbstract.mockQuerySkuDetails(querySkuDetailsSubsReturn = listOf(storeProduct))
+        mockBillingAbstract.mockQueryProductDetails(queryProductDetailsSubsReturn = listOf(storeProduct))
 
         onActivityReady { activity ->
-            Purchases.sharedInstance.purchaseProductWith(
-                activity = activity,
-                storeProduct = storeProduct,
+            Purchases.sharedInstance.purchaseWith(
+                purchaseParams = PurchaseParams.Builder(activity, storeProduct).build(),
                 onError = { error, _ -> fail("Purchase should be successful. Error: ${error.message}") },
                 onSuccess = { transaction, customerInfo ->
                     assertThat(transaction).isEqualTo(storeTransaction)
-                    assertThat(customerInfo.allPurchaseDatesByProduct.containsKey(storeProduct.sku)).isTrue
+                    assertThat(customerInfo.allPurchaseDatesByProduct.size).isEqualTo(1)
+                    val productId = customerInfo.allPurchaseDatesByProduct.keys.first()
+                    // Uncomment to check for the correct product id once changes in load shedder to include
+                    // base plan id are merged
+                    // val expectedProductId = "${Constants.productIdToPurchase}:${Constants.basePlanIdToPurchase}"
+                    // assertThat(productId).isEqualTo(expectedProductId)
```

Comment:
This is due to an issue in the load shedder, because it doesn't include the plan id in the response, which causes `allPurchaseDatesByProduct` to have different keys. Need to revisit this and add more checks once this and the fix for granting entitlements for new product ids are done in the load shedder.

Reply:
Created SDK-3050